sopel/tools/time.py

```diff
@@ -40,7 +55,7 @@ def validate_timezone(zone):
     becomes ``UTC``. In the majority of user-facing interactions, such
     case-insensitivity will be expected.
 
-    If the zone is not valid, ``ValueError`` will be raised.
+    If the zone is not valid, :exc:`ValueError` will be raised.
```

Comment:
Arguably, an invalid zone should raise `ValueError` too instead of returning `None`.

We can make this change in 8.0, as it's a major version bump.

Reply:
What do you mean by "an invalid zone"? Unless it's `None`, only valid zone will be returned, invalid zone will trigger a `ValueError`.

I could modify that to raise a `TypeError` if `zone is None`, if that's what you are thinking about.

Reply:
Think I managed to leave out some of my thought process, and tag the wrong lines. 😦

Seems kinda silly to return `None` if `zone is None`, when the defined semantics for "the zone is not valid" (`None` can never be valid) are `raise ValueError`. Why is `None` as input treated differently? It means having to add (as you did in this patch) `@typing.overload` definitions to special-case the return value of `None` that never happens otherwise.

The obvious change is always either return `str` or raise `ValueError`. I think removing the overloads and using `validate_timezone(zone: Optional[str])` would be OK. Just let `validate_timezone(None)` raise `ValueError` like any other invalid zone name would.

Searching internal usage of `validate_timezone()` yielded only a few call sites, one of which has a comment saying why `None` _not_ raising an exception like any other invalid value would is inconvenient:

https://github.com/sopel-irc/sopel/blob/e5bcd1fbfbc781eee990b47c0236f0772ba13124/sopel/modules/remind.py#L429-L433

Reply:
Ha yes, this way it makes sense! I agree with you, and since you are OK with this breaking change, I'll add it.